cns/restserver/util.go

```diff
@@ -923,3 +923,62 @@ func (service *HTTPRestService) createNetworkContainers(createNetworkContainerRe
 		Message:    "",
 	}
 }
+
+// setResponse encodes the http response
+func (service *HTTPRestService) setResponse(w http.ResponseWriter, returnCode types.ResponseCode, response interface{}) {
+	serviceErr := service.Listener.Encode(w, &response)
+	logger.Response(service.Name, response, returnCode, serviceErr)
+}
+
+// generateAndCacheGetHomeAzResponse build the GetHomeAzResponse and cache it
+func (service *HTTPRestService) generateAndCacheGetHomeAzResponse(returnMessage string, returnCode types.ResponseCode, homeAzResponse nmagent.HomeAzResponse) cns.GetHomeAzResponse {
+	getHomeAzResponse := cns.GetHomeAzResponse{
+		HomeAzResponse: homeAzResponse,
+		Response:       cns.Response{ReturnCode: returnCode, Message: returnMessage},
+	}
+	service.updateGetHomeAzResponseCache(getHomeAzResponse)
+	return getHomeAzResponse
+}
+
+// readNMASupportedApisCache gets the nmsSupportedApisCache value
+func (service *HTTPRestService) readNMASupportedApisCache() []string {
+	service.nmaSupportedApisCacheMux.RLock()
+	supportedApis := service.nmaSupportedApisCache
+	supportedApisCopy := make([]string, len(supportedApis))
+	copy(supportedApisCopy, supportedApis)
+	service.nmaSupportedApisCacheMux.RUnlock()
+	return supportedApisCopy
+}
+
+// updateNMASupportedApisCache updates the nmaSupportedApisCache value
+func (service *HTTPRestService) updateNMASupportedApisCache(supportedApis []string) {
+	service.nmaSupportedApisCacheMux.Lock()
+	service.nmaSupportedApisCache = supportedApis
+	service.nmaSupportedApisCacheMux.Unlock()
+}
+
+// readGetHomeAzResponseCache gets the GetHomeAzResponseCache value
+func (service *HTTPRestService) readGetHomeAzResponseCache() cns.GetHomeAzResponse {
+	service.getHomeAzResponseCacheMux.RLock()
+	response := service.getHomeAzResponseCache
+	service.getHomeAzResponseCacheMux.RUnlock()
+	return response
+}
+
+// updateGetHomeAzResponseCache updates the GetHomeAzResponseCache value
+func (service *HTTPRestService) updateGetHomeAzResponseCache(response cns.GetHomeAzResponse) {
+	service.getHomeAzResponseCacheMux.Lock()
+	service.getHomeAzResponseCache = response
+	service.getHomeAzResponseCacheMux.Unlock()
+}
```

Comment:
`*HTTPRestService` shouldn't really care that responses for these endpoints are cached. The only thing it cares about is that there is a thing that can get supported APIs and that can also get this HomeAz stuff. This can be better done by composing the client with a new `CachedClient`, and depending on the behavior expected from an NMAgent instead of a concrete implementation. I sketched this out here (apologies for the screencap, it was the only way I could get a decent example without making something contrived):

![image](https://user-images.githubusercontent.com/288061/199803401-c1a55413-302e-4aa2-8bbf-413de2f8b114.png)